db/version_set.cc

```diff
@@ -2618,6 +2618,13 @@ void VersionStorageInfo::AddFile(int level, FileMetaData* f, Logger* info_log) {
 #endif
   f->refs++;
   level_files->push_back(f);
+
+  const uint64_t file_number = f->fd.GetNumber();
+
+  FileLocation& file_location = file_locations_[file_number];
+  assert(!file_location.IsValid());
```

Comment:
Should we assert `file_locations_.find(file_number) == file_locations_.end()`?

Reply:
Sure, that also works. Will change.